src/main/java/de/torui/coflsky/WSCommandHandler.java

```diff
@@ -30,190 +31,208 @@
-		}
-		
-		if(cmd.startsWith("/cofl") || cmd.startsWith("http")) {
-			ClientCommandHandler.instance.executeCommand(sender, cmd);
-		} else {
-			Minecraft.getMinecraft().thePlayer.sendChatMessage(cmd);
-		}
-	}
-
-	
-	private static IChatComponent CommandToChatComponent(ChatMessageData wcmd) {
-		if(wcmd.OnClick != null)
-			lastOnClickEvent = "/cofl callback " + wcmd.OnClick;
-		if (wcmd.Text != null) {
-			IChatComponent comp = new ChatComponentText(wcmd.Text);
-
-			ChatStyle style;
-			if (wcmd.OnClick != null) {
-				if (wcmd.OnClick.startsWith("http")) {
-					style = new ChatStyle().setChatClickEvent(new ClickEvent(Action.OPEN_URL, wcmd.OnClick));
-				} else {
-					style = new ChatStyle()
-							.setChatClickEvent(new ClickEvent(Action.RUN_COMMAND, "/cofl callback " + wcmd.OnClick));
-				}
-				comp.setChatStyle(style);
-			}
-
-			if (wcmd.Hover != null && !wcmd.Hover.isEmpty()) {
-				if (comp.getChatStyle() == null)
-					comp.setChatStyle(new ChatStyle());
-				comp.getChatStyle().setChatHoverEvent(
-						new HoverEvent(HoverEvent.Action.SHOW_TEXT, new ChatComponentText(wcmd.Hover)));
-			}
-			return comp;
-		}
-		return null;
-	}
-	
-	private static void ChatMessage(Command<ChatMessageData[]> cmd) {		
-		ChatMessageData[] list = cmd.getData() ;//WSClient.gson.fromJson(cmd.getData(), WriteToChatCommand[].class);
-
-		IChatComponent master = new ChatComponentText("");
-
-		for (ChatMessageData wcmd : list) {
-			IChatComponent comp = CommandToChatComponent(wcmd);
-			if (comp != null)
-				master.appendSibling(comp);
-		}
-		Minecraft.getMinecraft().thePlayer.addChatMessage(master);
-	}
-
-	
-
-	private static void WriteToChat(Command<ChatMessageData> cmd) {
-		ChatMessageData wcmd = cmd.getData();
-		
-		IChatComponent comp = CommandToChatComponent(wcmd);
-		if (comp != null)
-		{
-			Minecraft.getMinecraft().thePlayer.addChatMessage(comp);
-		}
-			
-	}
+    public static transient String lastOnClickEvent;
+    public static FlipHandler flipHandler = new FlipHandler();
+    private static final ModListData modListData = new ModListData();
+    private static final Gson gson = new Gson();
+    private static final ProxyManager proxyManager = new ProxyManager();
+
+    public static boolean HandleCommand(JsonStringCommand cmd, Entity sender) {
+        // Entity sender = Minecraft.getMinecraft().thePlayer;
+        System.out.println("Handling Command=" + cmd.toString());
+
+        switch (cmd.getType()) {
+            case WriteToChat:
+                WriteToChat(cmd.GetAs(new TypeToken<ChatMessageData>() {
+                }));
+                break;
+            case Execute:
+                Execute(cmd.GetAs(new TypeToken<String>() {
+                }), sender);
+                break;
+            case PlaySound:
+                SoundData sc = cmd.GetAs(new TypeToken<SoundData>() {
+                }).getData();
+
+                PlaySound(sc.Name, sc.Pitch);
```

Comment:
```suggestion
                PlaySound(sc.Name, sc.Pitch);
```
There are no empty lines in the other commands, why here?

Reply:
Because this is the only command where it `getData` is invoced and stored in a variable because the `PlaySound` method takes 2 parameters as it gets reused on different places.
I can remove the emty line if you want though